2017-10-26  Lauren Jeanne Fromm  <jeanne@lnxsrv07.seas.ucla.edu>

	    * timespec.h: Replace comparison method in timespec_cmp with direct comparisons instead of subtraction to avoid integer overflow.


--- /u/cs/ugrad/jeanne/cs35l/lab4/coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ /u/cs/ugrad/jeanne/cs35l/lab4/timespec.h	2017-10-26 20:28:41.090385000 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec < b.tv_sec)
+    return -1;
+  else if (a.tv_sec > b.tv_sec)
+    return 1;
+  else
+    return 0;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
